python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -111,63 +119,14 @@ def run_iteration(
             yield
 
             asset_keys = check.not_none(run_request.asset_selection)
-            check.invariant(len(asset_keys) > 0)
-
-            repo_handle = asset_graph.get_repository_handle(asset_keys[0])
-
-            # Check that all asset keys are from the same repo
-            for key in asset_keys[1:]:
-                check.invariant(repo_handle == asset_graph.get_repository_handle(key))
-
-            location_name = repo_handle.code_location_origin.location_name
-            repository_name = repo_handle.repository_name
-            job_name = check.not_none(asset_graph.get_implicit_job_name_for_assets(asset_keys))
-
-            code_location = workspace.get_code_location(location_name)
-            external_job = code_location.get_external_job(
-                JobSubsetSelector(
-                    location_name=location_name,
-                    repository_name=repository_name,
-                    job_name=job_name,
-                    op_selection=None,
-                    asset_selection=asset_keys,
-                )
-            )
 
-            tags = {
-                **run_request.tags,
-                AUTO_MATERIALIZE_TAG: "true",
-                **instance.auto_materialize_run_tags,
-            }
-
-            external_execution_plan = code_location.get_external_execution_plan(
-                external_job,
-                run_request.run_config,
-                step_keys_to_execute=None,
-                known_state=None,
-                instance=instance,
+            run = submit_asset_run(
+                run_request,
+                instance,
+                workspace,
+                asset_graph,
+                {AUTO_MATERIALIZE_TAG: "true", **instance.auto_materialize_run_tags},
```

Comment:
No need to handle for this PR, but theoretically the auto-observe assets might desire a different tag for their runs (likely all auto-kicked off runs should have the same tag, but that tag shouldn't be called "auto_materialize")

Reply:
Hmm good point. I'm going to actually add an auto-observe tag, because this is weird otherwise.